File: 75 days Challenge Leetcode/PrefixSum/Day-19/Solution.py
from typing import List

class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        total_sum = sum(nums) # Calculate the total sum of all elements
        left_sum = 0 # Initialize left sum as 0
        
        # Iterate over the array and check if left sum equals right sum at each index
        for i in range(len(nums)):
            # If left sum equals right sum, return the current index
            if left_sum == total_sum - left_sum - nums[i]: 
                return i
            # Update left sum by adding the current element
            left_sum += nums[i]
        
        # If no pivot index is found, return -1
        return -1

File: 75 days Challenge Leetcode/PrefixSum/Day-19/Solution.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        int totalSum = 0, leftSum = 0;
        
        // Calculate the total sum of all elements in the array
        for (int num : nums) totalSum += num;
        
        // Iterate over the array and check if left sum equals right sum at each index
        for (int i = 0; i < nums.size(); i++) {
            // If left sum equals right sum, return the current index
            if (leftSum == totalSum - leftSum - nums[i])  
                return i;
            // Update left sum by adding the current element
            leftSum += nums[i];
        }
        
        // If no pivot index is found, return -1
        return -1;
    }
};
